### PR TITLE
Stamp origin with app on create

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
Ensure origin is stamped on new documents both on create and updateDocuments.

Fixed some issues with getting by Token.

### CODE REVIEW

File: src/services/penaltyDocuments.js [hunk 1]
```diff
@@ -44,14 +44,20 @@ export default class PenaltyDocument {
 		const dbGet = this.db.get(params).promise();
 
 		dbGet.then((data) => {
-			if (!data.Item) {
-				callback(null, createResponse(404, 'ITEM NOT FOUND'));
+			console.log(JSON.stringify(data.Item, null, 2));
+			console.log('before object check');
+			if (!data.Item || this.isEmpty(data.Item)) {
+				console.log('404 error here we come');
```
Comment (fabiosimaobjss) [inline src/services/penaltyDocuments.js]: @rossfauldsbjss  Not sure if you meant to leave this console.log, I like the funny message!

File: src/services/penaltyDocuments.js [hunk 2]
```diff
@@ -68,6 +74,10 @@ export default class PenaltyDocument {
 		});
 	}
 
+	isEmpty(obj) {
+		return JSON.stringify(obj) === JSON.stringify({});
```
Comment (fabiosimaobjss) [inline src/services/penaltyDocuments.js]: Just for reference, there's a library called lodash that has a isEmpty method (and plenty of other really useful methods) that does this in a more efficient manner.

File: src/services/penaltyDocuments.js [hunk 3]
```diff
@@ -120,19 +130,19 @@ export default class PenaltyDocument {
 		delete Value.paymentAuthCode;
 		delete Value.paymentDate;
 		// may not need to remove this delete Value.paymentToken;
+		if (typeof body.Origin === 'undefined') {
```
Comment (fabiosimaobjss) [inline src/services/penaltyDocuments.js]: This is exactly the same as if (body,Origin === undefined)

File: src/services/penaltyDocuments.js (hunk 1 above)
Comment (rossfauldsbjss) [reply src/services/penaltyDocuments.js]: will clean up console logs.

**Pull request merged**